R/pkratio-datamapping.R

```diff
@@ -9,30 +9,86 @@ PKRatioDataMapping <- R6::R6Class(
     sizeGrouping = NULL,
     shapeGrouping = NULL,
-
+    
+    colorGroupingDataFrame = NULL,
+    sizeGroupingDataFrame = NULL,
+    shapeGroupingDataFrame = NULL,
+    
+    colorLegendTitle = NULL,
+    sizeLegendTitle = NULL,
+    shapeLegendTitle = NULL,
+    
     # Example of how to do some other stuff
     initialize = function(x = "Age",
                           y = "Ratio",
                           colorGrouping = NULL,
                           sizeGrouping = NULL,
-                          shapeGrouping = NULL) {
-      self$colorGrouping <- colorGrouping
-      self$sizeGrouping <- sizeGrouping
-      self$shapeGrouping <- shapeGrouping
-
+                          shapeGrouping = NULL,
+                          colorGroupingDataFrame = NULL,
+                          sizeGroupingDataFrame = NULL,
+                          shapeGroupingDataFrame = NULL,
+                          colorLegendTitle = NULL,
+                          sizeLegendTitle = NULL,
+                          shapeLegendTitle = NULL) {
+      
+      colorGroupingList <- self$get_grouping_list(colorGrouping,colorGroupingDataFrame)
+      sizeGroupingList  <- self$get_grouping_list(sizeGrouping,sizeGroupingDataFrame)
+      shapeGroupingList <- self$get_grouping_list(shapeGrouping,shapeGroupingDataFrame)
+      
+      self$colorGrouping <- colorGroupingList$selfGrouping
+      self$sizeGrouping  <- sizeGroupingList$selfGrouping
+      self$shapeGrouping <- shapeGroupingList$selfGrouping
+      
+      self$colorGroupingDataFrame <- colorGroupingList$selfGroupingDataFrame
+      self$sizeGroupingDataFrame  <- sizeGroupingList$selfGroupingDataFrame
+      self$shapeGroupingDataFrame <- shapeGroupingList$selfGroupingDataFrame
+      
+      self$colorLegendTitle = colorLegendTitle
+      self$sizeLegendTitle = sizeLegendTitle
+      self$shapeLegendTitle = shapeLegendTitle
+      
       super$initialize(x, y)
     },
-
+    
+    
+    get_grouping_list = function(Grouping,GroupingDataFrame){  #move out of public?
+      if (!is.null(GroupingDataFrame)){
+        selfGroupingDataFrame <- GroupingDataFrame
+        selfGrouping<-tail(colnames(GroupingDataFrame),1)
+      } else {
+        selfGroupingDataFrame <- NULL
+        selfGrouping <- Grouping
+      }
+      groupinglist <-list(selfGroupingDataFrame = selfGroupingDataFrame , selfGrouping = selfGrouping)
+      return(groupinglist)
+    },
+    
+    
     getGrouping = function(data, metaData) {
-      data$colorGrouping <- getDefaultCaptions(data = data, metaData = metaData, variableList = self$colorGrouping)
-      data$sizeGrouping <- getDefaultCaptions(data = data, metaData = metaData, variableList = self$sizeGrouping)
-      data$shapeGrouping <- getDefaultCaptions(data = data, metaData = metaData, variableList = self$shapeGrouping)
-
+      
+      markerTypeList = list( 
+        list(Grp = self$colorGrouping , Df = self$colorGroupingDataFrame , DefaultColName = "colorGrouping" ),

```

Comment:
same with variable name in the list. Lower case => grp, df, defaultColName etc..